src/redist/targets/GenerateLayout.targets

```diff
@@ -492,6 +492,14 @@
 
   </Target>
 
+  <Target Name="GeneratePrecomputedRarCache" DependsOnTargets="LayoutTemplates;LayoutManifests;LayoutBundledTools">
+    <ItemGroup>
+      <SampleProjectForResolveAssemblyReferenceCache Include="$(RepoRoot)\TestAssets\TestProjects\RARPreCacheProject\**\*.*" />
+    </ItemGroup>
+    <Copy SourceFiles="@(SampleProjectForResolveAssemblyReferenceCache)" DestinationFiles="$(RepoRoot)\artifacts\bin\redist\$(Configuration)\RARPreCacheProject\%(RecursiveDir)%(FileName)%(Extension)" />
+    <Exec Command="$(RepoRoot)\artifacts\bin\redist\$(Configuration)\dotnet\dotnet.exe msbuild $(RepoRoot)\artifacts\bin\redist\$(Configuration)\RARPreCacheProject\RARPreCacheProject.csproj /t:ResolveAssemblyReferences /p:AssemblyInformationCacheOutputPath=$(SdkInternalLayoutPath)sdk\$(Version)\SDKPrecomputedAssemblyReferences.cache" />
```

Comment:
There should be a better existing property for $(RepoRoot)\artifacts\bin\redist\$(Configuration)\dotnet\dotnet.exe  (stage2 SDK location)

And you should only run this on a windows leg

Reply:
Not you mention it, I think it's equivalent to $(SdkInternalLayoutPath)dotnet.exe. I can change it to that.

Why only on a windows leg? I think it should work for any OS.